### ISSUE TITLE
Is PHP 7 Supported?

### ISSUE THREAD

**Issue by jerrac** [iionly/tidypics#39]
When testing TidyPics in a php:apache docker container, I get this error:

```
web_1    | [Thu Dec 21 04:43:58.142118 2017] [php7:error] [pid 26] [client 172.23.0.1:56470] PHP Fatal error:  Declaration of TidypicsAlbum::delete() must be compatible with ElggEntity::delete($recursive = true) in /var/www/html/mod/tidypics/classes/TidypicsAlbum.php on line 0, referer: http://localhost:8000/photos/add/42
web_1    | 172.23.0.1 - - [21/Dec/2017:04:43:58 +0000] "POST /action/photos/album/save HTTP/1.1" 500 319 "http://localhost:8000/photos/add/42" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0"
```
I tried a couple different releases: https://elgg.org/plugins/385077/releases/2553279/download and https://elgg.org/plugins/385077/releases/2595533/download

When I switch the image to php:5-apache, the problem goes away.

Judging from https://github.com/Elgg/Elgg/pull/11434 Elgg core is moving to php 7. But I haven't dug very deep into it yet, so maybe it isn't quite there?

Anyway, apologies if this question should have gone elsewhere. I'm brand new to the Elgg community, so I haven't figured out all the ropes yet. :)

**Comment by iionly** [comment]
The latest release of Tidypics should work on PHP 7.0 and 7.1. The error should be on PHP 7.2 only.

I fixed the PHP 7.2 issue with Tidypics here in the master branch but had no time yet to make a new release. So, if you need Tidypics to work with PHP 7.2 now you could download the master branch from here and it should work with Elgg 2.3.x. I think I won't manage to make a new release this year anymore but hopefully early next year.

The upcoming Elgg 3 will require PHP 7 but Elgg 2.3 will still work with PHP 5.6. But there will be a new and updated version of Tidypics necessary for Elgg 3 due to all the changes that come with Elgg 3. I hope I can start working on that soon but it's still a bit too early as Elgg 3 is still in the works.

**Comment by jerrac** [comment]
Awesome. Thanks!
